x/crosschain/keeper/keeper_cross_chain_tx_vote_inbound_tx.go

```diff
@@ -85,30 +85,59 @@ func (k msgServer) VoteOnObservedInboundTx(goCtx context.Context, msg *types.Msg
 
 	// Inbound Ballot has been finalized , Create CCTX
 	cctx := k.CreateNewCCTX(ctx, msg, index, types.CctxStatus_PendingInbound, observationChain, receiverChain)
+	defer func() {
+		EmitEventInboundFinalized(ctx, &cctx)
+		k.SetCrossChainTx(ctx, cctx)
+	}()
 	// FinalizeInbound updates CCTX Prices and Nonce
 	// Aborts is any of the updates fail
-
 	if receiverChain.IsZetaChain() {
-		err = k.HandleEVMDeposit(ctx, &cctx, *msg, observationChain)
-		if err != nil {
+		tmpCtx, commit := ctx.CacheContext()
```

Comment:
maybe we can a cli test to x/client/test util to verify the commit()
Not needed for this PR though , can be added later